### PR TITLE
✨ Remove spaces from machine-readable comments

### PR DESCRIPTION
**Pull request by Adirio**
# Description
Machine-readable comments used for markers have spaces after the comment character (`//` or `#`), they have been removed to follow common coding practices.

Closes: #1384

# Comments to reviewers
131/144 files are from the `testdata` directory, this PR is smaller than what it looks like by the number of files changed.

### CODE REVIEW

**Comment by camilamacedo86** [comment]
Hi @Adirio, 

It is not passing in the CI. Could you please check this one for we get it merged. 
Shows that it is missing run make generate only.

**Comment by Adirio** [comment]
I think it is a rebase what it is needed. Will try it.

**Review by estroz — COMMENTED** [review]
Something like this can be done for any plugin, stable/frozen or not, since machine-readable comments with spaces are backwards compatible.

**Comment by camilamacedo86** [comment]
Hi @Adirio 

What was your motivation to define that #1384 (this change) would be a braking change and that should be done only for v3? If it is backwards compatible with v2 then I agree with @estroz in https://github.com/kubernetes-sigs/kubebuilder/pull/1868#pullrequestreview-550677466. 

However, i imagine that you saw some scenario that would be affected. so, could you please clarify?

**Comment by Adirio** [comment]
If we apply this to v2, projects scaffolded with v2 before applyign this change will have spaces and further update operation by kubebuilder will not be able to find the placeholders. For example the ones in `mian.go` to register the controllers in the manager.

**Comment by estroz** [comment]
> further update operation by kubebuilder will not be able to find the placeholders

If that’s true, it sounds like a bug in controller-tools’ comment lexer. Both forms are valid marker comments so should be parsed in the same manner.

**Comment by Adirio** [comment]
I mean the comments that are meant for kubebuilder itself, for example the `MainUpdater` inserts code fragments to register new controllers. Those fragments are inserted just before a machine-readabale comment and it does use exact match.

**Comment by estroz** [comment]
What about creating a method to compare marker comments like so

```go
func (m Marker) EqualsLine(line string) bool {
	line = strings.TrimSpace(strings.TrimPrefix(m.comment))
	return line == m.value
}
```

and use it [here](https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/internal/machinery/scaffold.go#L346):

```diff
		line := scanner.Text()
		for marker, codeFragments := range codeFragmentsMap {
-			if strings.TrimSpace(line) == strings.TrimSpace(marker.String()) {
+			if marker.EqualsLine(line) {
				...
			}
		}
```

**Comment by estroz** [comment]
The above is just a suggestion btw. Your solution also works. Feel free to unhold to merge as-is.

/lgtm
/hold

**Comment by Adirio** [comment]
/cancel hold

@estroz I liked your suggestion, removed the hold as the `lgtm` label is required again

File: pkg/model/file/marker.go [hunk 1]
```diff
@@ -19,15 +19,14 @@ package file
...
 const prefix = "+kubebuilder:scaffold:"
 
 var commentsByExt = map[string]string{
-	// TODO(v3): machine-readable comments should not have spaces by Go convention. However,
-	//  this is a backwards incompatible change, and thus should be done for next project version.
-	".go":   "// ",
-	".yaml": "# ",
+	".go":   "//",
+	".yaml": "#",
 	".yml":  "#",
```
Comment (camilamacedo86) [inline pkg/model/file/marker.go:30]: Tks 👍 for addressing : https://github.com/kubernetes-sigs/kubebuilder/pull/1907

File: pkg/plugins/internal/machinery/scaffold.go [hunk 2]
```diff
@@ -343,7 +343,7 @@ func insertStrings(content string, codeFragmentsMap file.CodeFragmentsMap) ([]by
 		line := scanner.Text()
 
 		for marker, codeFragments := range codeFragmentsMap {
-			if strings.TrimSpace(line) == strings.TrimSpace(marker.String()) {
+			if marker.EqualsLine(line) {
```
Comment (camilamacedo86) [inline pkg/plugins/internal/machinery/scaffold.go:346]: if we added here || strings.TrimSpace(line) == strings.TrimSpace(marker.String()) would not solve for both scenarios?

**Comment by camilamacedo86** [comment]
Hi @Adirio,

it is not passing in the CI.  Maybe it is missing a rebase?

**Comment by Adirio** [comment]
I don't know why but seems that one of the changes in `*_webhook.go` files seems to be breaking the webhooks. @estroz do you know if this should be considered a parsing error from the side of controller-tools?

**Comment by Adirio** [comment]
The part that is failing is this one: https://github.com/kubernetes-sigs/kubebuilder/blob/d501fc35a6deed5aac41b116a0f26182d85d99f3/testdata/project-v3/api/v1/webhook_suite_test.go#L126-L136

I'm not modifying anything that should break that check.

**Review by prafull01 — COMMENTED** [review]
I have looked into the code and the reason for this failure is following scaffolding code is not generated by kubebuilder:
```
err = (&Captain{}).SetupWebhookWithManager(mgr)
	Expect(err).NotTo(HaveOccurred())
```
Attaching the screenshot of failure and missing piece of code from scaffold code.
Screenshot from 2020-12-17 01-31-38
Screenshot from 2020-12-17 01-31-48

File: testdata/project-v3/api/v1/webhook_suite_test.go [hunk 3]
```diff
@@ -114,7 +114,7 @@ var _ = BeforeSuite(func() {
 	err = (&Captain{}).SetupWebhookWithManager(mgr)
 	Expect(err).NotTo(HaveOccurred())
```
Comment (Adirio) [inline testdata/project-v3/api/v1/webhook_suite_test.go:115]: It is scaffolded here, right?

Comment (prafull01) [reply testdata/project-v3/api/v1/webhook_suite_test.go:115]: I think you have not run make generate before pushing the change. I cloned your repo and generated a project with that, it doesn't generate that code. Also, I tried to run the `make generate` locally and it is throwing same error. Could you please try that?

**Review by prafull01 — APPROVED** [review]

**Review by estroz — APPROVED** [review]
/approve
/lgtm

**Pull request merged**